packages/backend-modules/discussions/graphql/resolvers/Comment.js

```diff
@@ -227,67 +227,51 @@ module.exports = {
       profilePicture: null,
       anonymity: true,
       username: null,
+      slug: null,
     }
 
-    if (!comment.userId) {
+    if (!commenter) {
       return anonymousComment
     }
 
-    const [discussion, commenter, commenterPreferences] = await Promise.all([
-      loaders.Discussion.byId.load(comment.discussionId),
-      loaders.User.byId.load(comment.userId),
-      loaders.Discussion.Commenter.discussionPreferences.load({
-        userId: comment.userId,
-        discussionId: comment.discussionId,
-      }),
-    ])
-
-    const credential = commenterPreferences && commenterPreferences.credential
-
-    let anonymous
-    if (discussion.anonymity === 'ENFORCED') {
-      anonymous = true
-    } else {
-      // FORBIDDEN or ALLOWED
-      if (commenterPreferences && commenterPreferences.anonymous != null) {
-        anonymous = commenterPreferences.anonymous
-      } else {
-        anonymous = false
+    const { discussion, commenterPreferences } = await Promise.props({
+      discussion: loaders.Discussion.byId.load(discussionId),
+      commenterPreferences:
+        loaders.Discussion.Commenter.discussionPreferences.load({
+          userId,
+          discussionId,
+        }),
+    })
```

Comment:
Does Promise.props do the same thing as
```
const [discussion, commenterPreferences] = await Promise.all([..., ...])
```
??

Reply:
As far as I understand the [documentation](http://bluebirdjs.com/docs/api/promise.props.html) it does: 

> Like [.all](http://bluebirdjs.com/docs/api/all.html) but for object properties or Maps* entries instead of iterated values. Returns a promise that is fulfilled when all the properties of the object or the Map's' values** are fulfilled.

But I agree with Henning that it's probably not worth to add the bluebird dependency for this.